business/definitionService.js

```diff
@@ -237,6 +250,25 @@ class DefinitionService {
     return definition
   }
 
+  /**
+   * Obtains an int32 hash representing the passed-in coordinates.
+   *
+   * @param {Coordinates} coordinates - individual or array of coordinates to obtain a hash for
+   */
+  tagFromCoordinates(coordinates) {
+    let result
+    if (Array.isArray(coordinates)) {
+      return coordinates.map(item => this.tagFromCoordinates(item.coordinates))
+        .filter((v, i, a) => v && a.indexOf(v) === i)
+        .join(',')
+    }
+    else if (coordinates) {
+      let hashBase = [coordinates.type, coordinates.name, coordinates.revision].join('|')
+      result = stringHash(hashBase).toString()
+    }
+    return result
```

Comment:
I think we should move the tag concept into the `Cdn` class and just pass coordinates around.

the tag is very cloudflare specific in this case

above would be:
`this.cdn.invalidate(coordinates)`

What do you think?

Reply:
You’re right that it is specific to content delivery, but not particularly to Cloudflare as you can see here https://developer.akamai.com/blog/2019/03/25/introducing-ability-purge-cache-tag

Reply:
interesting - thanks for the link 👍